src/object/obj_class.c

```diff
@@ -669,12 +616,174 @@ oclass_ident2cl(daos_oclass_id_t oc_id)
 
 	idx = daos_array_find(oc_ident_array, oc_ident_array_sz, oc_id,
 			      &oc_ident_sort_ops);
+	if (idx >= 0) {
+		if (nr_grps)
+			*nr_grps = oc_ident_array[idx]->oc_grp_nr;
+
+		return oc_ident_array[idx];
+	}
+
+	if (oc_id < (OR_RP_1 << OC_REDUN_SHIFT))
+		return NULL;
+
+	idx = daos_array_find(oc_ident_array, oc_ident_array_sz, oc_id,
```

Comment:
Please add some comments for this, I struggled a while before I can understand this

Reply:
Yup..this is tricky.